ambry-api/src/main/java/com/github/ambry/rest/RequestPath.java

```diff
@@ -298,4 +312,30 @@ private static int matchPathSegments(String path, int pathOffset, String pathSeg
     }
     return nextPathSegmentOffset;
   }
+
+  /**
+   * Get named blob path from S3 request path
+   * @param path s3 request path
+   * @return named blob request path
+   */
+  private static String convertToNamedBlobPath(String path) {
+    // S3 requests to Ambry are in the form "/s3/account-name/key-name". We convert it to named blob path
+    // "/named/account-name/container-name/key-name" internally.
+    // For ex: for S3 path, "/s3/named-blob-sandbox/checkpoints/87833badf879a3fc7bf151adfe928eac/chk-1/_metadata",
+    // the corresponding named blob path is "/named/named-blob-sandbox/container-a/checkpoints/87833badf879a3fc7bf151adfe928eac/chk-1/_metadata".
+    // Please note that we hardcode container-name to 'container-a'.
+    int accountStart = S3_PATH.length() + PATH_SEPARATOR_STRING.length();
+    int accountEnd = path.indexOf(PATH_SEPARATOR_CHAR, accountStart);
+    if (accountEnd == -1) {
+      accountEnd = path.length();
+    }
```

Comment:
this seems to be a 404 issue. If request goes like "/s3/", then we should return 404.